tests/components/homekit/test_type_switches.py

```diff
@@ -387,3 +387,46 @@ async def test_script_switch(hass, hk_driver, events):
     await hass.async_block_till_done()
     assert acc.char_on.value is False
     assert len(events) == 1
+
+
+async def test_select_switch(hass, hk_driver, events):
+    """Test if select switch accessory is reset correctly."""
+    domain = "input_select"
+    entity_id = "input_select.test"
+
+    hass.states.async_set(entity_id, None)
+    await hass.async_block_till_done()
+    acc = SelectSwitch(hass, hk_driver, "Switch", entity_id, 2, None)
+    await acc.run()
+    await hass.async_block_till_done()
+
+    assert acc.char_on.value is False
+
+    call_turn_on = async_mock_service(hass, domain, "test")
+    call_turn_off = async_mock_service(hass, domain, "turn_off")
+
+    await hass.async_add_executor_job(acc.char_on.client_update_value, True)
+    await hass.async_block_till_done()
+    assert acc.char_on.value is True
+    assert call_turn_on
+    assert call_turn_on[0].data == {}
+    assert len(events) == 1
+    assert events[-1].data[ATTR_VALUE] is None
+
+    future = dt_util.utcnow() + timedelta(seconds=1)
+    async_fire_time_changed(hass, future)
+    await hass.async_block_till_done()
+    assert acc.char_on.value is True
+
+    future = dt_util.utcnow() + timedelta(seconds=10)
+    async_fire_time_changed(hass, future)
+    await hass.async_block_till_done()
+    assert acc.char_on.value is False
+
+    assert len(events) == 1
+    assert not call_turn_off
+
+    await hass.async_add_executor_job(acc.char_on.client_update_value, False)
```

Comment:
```suggestion
    acc.char_on.client_update_value(False)
```
`client_update_value` doesn't need to run in the executor anymore.  These are left over from when homekit wasn't async.